### PR TITLE
fix(pages-components): properly obfuscate links

### PR DESCRIPTION
**Pull request by mkilpatrick**
Before this change Links (`a` tags) were "obfuscated" simply by reversing the string and using unicodeBidi and LTR/RTL direction CSS. This caused issues for real RTL clients. Additionally, it didn't really obfuscate the actual email addresses. Now we use base64 encoding of the email and decode it on click.

Also fixes a bug that caused a DOM warning when `obfuscate` was set to true because it was being passed in the `rest` props to the `a` tag.

Now also handles not double prefixing tel: and mailto: if the user sets it themself in the link.

### CODE REVIEW

File: packages/pages-components/src/components/link/link.tsx [hunk 1]
```diff
@@ -46,26 +51,26 @@ export const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(
...
+        const encodedLink = currentTarget.href.split("/").at(-1);
+        if (encodedLink) {
+          window.location.href = atob(encodedLink);
+        }
+      }
     };
 
-    const renderedLink = isObfuscate ? reverse(link.link) : link.link;
+    const renderedLink = isObfuscate
+      ? "Obfuscated, set a label or child"
```
Comment (benmcginnis) [inline packages/pages-components/src/components/link/link.tsx:64]: should this get translated?

File: packages/pages-components/src/components/link/methods.ts [hunk 2]
```diff
@@ -24,10 +30,14 @@ const getHref = (cta: CTA): string => {
  * Regex defined in HTML Spec for <input type="email"> validation.
  * https://html.spec.whatwg.org/#email-state-(type=email)
  */
-const isEmail = (string: string): boolean => {
+const isEmail = (link: string): boolean => {
+  if (link.includes("mailto:")) {
```
Comment (benmcginnis) [inline packages/pages-components/src/components/link/methods.ts:34]: probably a nit, but wouldn't this match `http://www.mysuperawesomewebsite?mailto:=foo`?

**Review by benmcginnis — COMMENTED** [review]
generally lgtm

File: packages/pages-components/src/components/link/link.tsx (hunk 1 above)
Comment (mkilpatrick) [reply packages/pages-components/src/components/link/link.tsx:64]: We don't have translations in any components. But either way, this is more meant for an indicator for the dev to set a label properly. Open to other suggestions though.

File: packages/pages-components/src/components/link/methods.ts (hunk 2 above)
Comment (mkilpatrick) [reply packages/pages-components/src/components/link/methods.ts:34]: Good call. Changed it to check using startsWith

File: packages/pages-components/src/components/link/link.tsx (hunk 1 above)
Comment (benmcginnis) [reply packages/pages-components/src/components/link/link.tsx:64]: ah ok. I feel like y'all probably want to remove any strings generally and have them be injectable

**Review by benmcginnis — APPROVED** [review]
lgtm

**Comment by mkilpatrick** [comment]
> @mkilpatrick this caused a very hard to debug issue on my end when updating the package version.
> 
> ```
> Error 500
> 500 TypeError: Cannot read properties of undefined (reading 'startsWith')
> ```
> 
> Could we the nullish coalescing operator here? I know the TS typings are disallowing this, but in some cases it makes sense to pass on an empty object.

Can you provide an example usage that results in the error?

**Comment by mkilpatrick** [comment]
@hbende4 I'm not sure there's a great way to handle that. `{}` doesn't satisfy the CTA type which is what the component expects. For most components you need to code defensively and only render it if the object being used is defined. In this case you should do:

```
{c_ctaField && <Link ...}
```
I could change it to throw. Would that be more helpful? The stacktrace won't be any better though because the library is bundled.
```
Error: CTA's link is undefined
at file:///Users/mkilpatrick/Desktop/rp-test/node_modules/@yext/pages-components/dist/index-bhUPnB8J.js:2134:13
at renderWithHooks (/Users/mkilpatrick/Desktop/rp-test/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:5662:16)
at renderForwardRef (/Users/mkilpatrick/Desktop/rp-test/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:5857:18)
...
```

**Comment by mkilpatrick** [comment]
Btw we now throw an error to make the stack trace easier to debug now in 1.0.2

**Comment by hbendev** [comment]
@mkilpatrick another thing that seems to be broken with the `<Link/>` component starting from version `1.0.0-rc.11`, likely due to this PR is `mailto:` links.

The below snippet works with v `1.0.0-rc.10`, but the element rendered does nothing when clicked on the newer version.

```tsx
{cta?.label && (
  <Link
    className=""
    eventName="my_cta"
    href={mailto:my@address.com}
  >
    {cta.label}
  </Link>
)}
 ```

**Comment by mkilpatrick** [comment]
Thanks for pointing that out @hbendev. I put up a fix and will release shortly. A workaround would be to use the `cta.link` param on the Link component with a

**Pull request merged**